### PR TITLE
OCPBUGS-44800: Add Dynamic Plugins nav item

### PR DESCRIPTION
**Pull request by rhamilto**
localhost_9000_k8s_cluster_operator openshift io~v1~Console_cluster_console-plugins

### CODE REVIEW

**Comment by rhamilto** [comment]
We use the label `Dynamic plugins` in the Overview and the about modal (screenshots below), but I think `Console Plugins` is the proper choice for the nav item since that aligns with the tab label, but it is odd we have this discrepancy.  Maybe we need to revisit?  @jhadvig, @spadgett, @vojtechszocs, WDYT?
Screenshot 2024-11-20 at 9 53 05 AM
Screenshot 2024-11-20 at 9 53 15 AM

**Comment by rhamilto** [comment]
> We use the label `Dynamic plugins` in the Overview and the About modal (screenshots below), but I think `Console Plugins` is the proper choice for the nav item since that aligns with the tab label, but it is odd we have this discrepancy. Maybe we need to revisit? @jhadvig, @spadgett, @vojtechszocs, WDYT?

In the light of a new day, I realized the nav item label should be `Dynamic Plugins`, so I've updated this PR.

File: frontend/packages/console-app/src/consts.ts [hunk 1]
```diff
@@ -5,3 +5,4 @@ export const HIDE_USER_WORKLOAD_NOTIFICATIONS_USER_SETTINGS_KEY =
 export const FLAG_DEVELOPER_PERSPECTIVE = 'DEVELOPER_PERSPECTIVE';
 export const ACM_PERSPECTIVE_ID = 'acm';
 export const ADMIN_PERSPECTIVE_ID = 'admin';
+export const FLAG_CAN_GET_CONSOLE_OPERATOR_CONFIG = 'FLAG_CAN_GET_CONSOLE_OPERATOR_CONFIG';
```
Comment (vojtechszocs) [inline frontend/packages/console-app/src/consts.ts:8]:
```suggestion
export const FLAG_CAN_GET_CONSOLE_OPERATOR_CONFIG = 'CAN_GET_CONSOLE_OPERATOR_CONFIG';
```

**Review by vojtechszocs — CHANGES_REQUESTED** [review]

File: frontend/packages/console-app/src/consts.ts (hunk 1 above)
Comment (rhamilto) [reply frontend/packages/console-app/src/consts.ts]: Good catch, @vojtechszocs.  Thanks.  Updated.

**Comment by rhamilto** [comment]
> > We use the label `Dynamic plugins` in the Overview and the About modal (screenshots below), but I think `Console Plugins` is the proper choice for the nav item since that aligns with the tab label, but it is odd we have this discrepancy. Maybe we need to revisit? @jhadvig, @spadgett, @vojtechszocs, WDYT?
> 
> IMO it should be "Console Plugins" everywhere. That's the only kind of plugin we've ever had an external API for. "Static plugins" were always an internal thing, so there's not a need to differentiate between them for users. I think putting "Dynamic" in the label raises more questions than it answers for most users. Also "Console Plugin" matches the CRD kind.

IIRC, we went with Dynamic Plugins so as not to conflate the functionality with the resource.  One could argue that subtlety is lost on users however...

**Comment by spadgett** [comment]
> IIRC, we went with Dynamic Plugins so as not to conflate the functionality with the resource. One could argue that subtlety is lost on users however...

To me, this could create more confusion because it sounds like they're two different things. There are also other kinds of plugins on the platform (for instance, network plugins or CLI plugins). The most important differentiator is that it's a _console_ plugin.
 
> Docs uses "Dynamic Plugins": https://docs.openshift.com/container-platform/4.17/web_console/dynamic-plugin/overview-dynamic-plugin.html

I'd update the docs as well to use "Console Plugins."

**Comment by rhamilto** [comment]
> https://docs.openshift.com/container-platform/4.17/web_console/dynamic-plugin/overview-dynamic-plugin.html

I added [a story for 4.19](https://issues.redhat.com/browse/CONSOLE-4364).

**Pull request merged**